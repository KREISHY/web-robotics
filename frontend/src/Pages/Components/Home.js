import { useState, useEffect } from 'react';
import {
  Box,
  Button,
  Flex,
  FormControl,
  FormLabel,
  Input,
  Stack,
  Avatar,
  Center,
} from "@chakra-ui/react";
import { Link } from 'react-router-dom';
import Login from './Login';
import AvatarProfile from './Avatar.jpg';

const Home = ({ currentUser, setCurrentUser, client }) => {
  const [email, setEmail] = useState('');
  const [firstname, setFirstname] = useState('');
  const [lastname, setLastname] = useState('');
  const [patronymic, setPatronymic] = useState('');

  useEffect(() => {
    const fetchUserData = async () => {
      if (currentUser) {
        const response = await client.get("/api/user/current-user/");
        setEmail(response.data.email);
        setFirstname(response.data.first_name);
        setLastname(response.data.last_name);
        setPatronymic(response.data.patronymic);
      }
    };

    fetchUserData();
  }, [currentUser, client]);

  return (
    <Box textAlign="center" mt="10">
      {!currentUser ? <Login setCurrentUser={setCurrentUser} /> : 
        <>
          <Flex minH={'100vh'} align={'center'} justify={'center'}>
            <Stack spacing={4} w={'full'} maxW={'md'} rounded={'xl'} boxShadow={'lg'} p={6} my={12}>
              <FormControl id="userName">
                <Center>
                  <Avatar size="xl" src={AvatarProfile}></Avatar>
                </Center>
              </FormControl>
              <FormControl id="email">
                <FormLabel>Email</FormLabel>
                <Input value={email} readOnly />
              </FormControl>
              <FormControl id="firstName">
                <FormLabel>Имя</FormLabel>
                <Input value={firstname} readOnly />
              </FormControl>
              <FormControl id="lastName">
                <FormLabel>Фамилия</FormLabel>
                <Input value={lastname} readOnly />
              </FormControl>
                <FormControl id="patronymic">
                <FormLabel>Отчество</FormLabel>
                <Input value={patronymic} readOnly />
              </FormControl>
              <Link to="/profile/change-data">
                <Button bg={'blue.400'} color={'white'} w="full" _hover={{ bg: 'blue.500' }}>Редактировать профиль</Button>
              </Link>
            </Stack>
          </Flex>
        </>
      }
    </Box>
  );
};

export default Home;