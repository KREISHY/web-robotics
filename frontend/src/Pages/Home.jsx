import './style.css';
import { Container, Row, Col, Button } from 'react-bootstrap';
import img from './new_img1.png';
import { useNavigate } from 'react-router-dom';

function Home() {
  const navigate =useNavigate();
  return (
    <div>
      <div>
        <Container className='tale'>
          <Row className='align-items-center text-center'>

              <h1 className=''>Добро пожаловать в "Дикция+"</h1>
              <span>
                «НАЗВАНИЕ» — это инновационное веб-приложение,
                созданное специально для логопедов, чтобы помочь
                вам в работе с пациентами, страдающими дизартрией.
              </span>
              <p><Button onClick={() => navigate('/tests/')}>Начать проходить тесты</Button></p>

          </Row>
        </Container>
      </div>
      <Container className='text-center tale'>
        <Row className='justify-content-center'>
          <Col xs={12} md={8}>
            <h1>Как начать?</h1>
            <ul className='list-unstyled home-list'>
              <li>Зарегистрируйтесь или войдите в свою учетную запись.</li>
              <li>Изучайте упражнения: Просматривайте каталог упражнений и выбирайте подходящие для ваших пациентов.</li>
              <li>Создавайте свои упражнения: Используйте нашу библиотеку материалов для создания уникальных занятий.</li>
              <li>Отслеживайте прогресс: Сохраняйте результаты и анализируйте успехи ваших пациентов.</li>
            </ul>
          </Col>
        </Row> 
      </Container>
    </div>
  );
}

export default Home;