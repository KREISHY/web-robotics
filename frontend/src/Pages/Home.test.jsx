import '@testing-library/jest-dom';
import { render, screen, waitFor } from '@testing-library/react';
import Home from './Home';
import { Img, list } from '@chakra-ui/react';

describe('Home component', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(<Home />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('Welcome renders', () => {
        render(<Home />);
        expect(screen.getByText('Добро пожаловать в "НАЗВАНИЕ"')).toBeInTheDocument();
    });

    it('Image is displayed', () => {
        render(<Home />);
        const image = screen.getByRole('img');
        expect(image).toBeInTheDocument();
        expect(image).toBeVisible();
    });

    it('Descripiton renders', () => {
        render(<Home />);
        expect(screen.getByText(/«НАЗВАНИЕ» — это инновационное веб-приложение/i)).toBeInTheDocument();
    });

    it('List renders', () => {
        render(<Home />);
        expect(screen.getByRole('list')).toBeInTheDocument();
    })
});