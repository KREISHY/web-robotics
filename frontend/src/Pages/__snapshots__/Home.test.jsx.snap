// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="color"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <img
              alt="изображения"
              src="new_img1.png"
              width="700px"
            />
          </div>
          <div
            class="col"
          >
            <h1
              class="hwhite"
            >
              Добро пожаловать в "НАЗВАНИЕ"
            </h1>
            <span
              class="white-text"
            >
              «НАЗВАНИЕ» — это инновационное веб-приложение, созданное специально для логопедов, чтобы помочь вам в работе с пациентами, страдающими дизартрией.
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col"
        />
        <div
          class="col"
        >
          <h1>
            Как начать?
          </h1>
          <ul>
            <li>
               Зарегистрируйтесь или войдите в свою учетную запись.
            </li>
            <li>
               Изучайте упражнения: Просматривайте каталог упражнений и выбирайте подходящие для ваших пациентов.
            </li>
            <li>
               Создавайте свои упражнения: Используйте нашу библиотеку материалов для создания уникальных занятий.
            </li>
            <li>
               Отслеживайте прогресс: Сохраняйте результаты и анализируйте успехи ваших пациентов.
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
